src/target/imxrt.c

```diff
@@ -227,6 +270,68 @@ bool imxrt_probe(target_s *const target)
 	return true;
 }
 
+static void imxrt_id_chip(target_s *target, imxrt_priv_s *priv)
+{
+	// The iMX RT series doesn't have a device id register. Instead, the NXP universal flash loader
+	// uses known ROM values at a particular address to differentiate the devices. That code uses
+	// three locations but only one location is actually needed.
+	// https://github.com/nxp-mcuxpresso/i.mxrt-ufl/blob/main/src/ufl_find_target.c
+	uint32_t rom_location = 0;
+	const uint16_t cpuid_partno = target->cpuid & CPUID_PARTNO_MASK;
+	if (cpuid_partno == CORTEX_M33)
+		rom_location = IMXRT_X00_ROM_FINGERPRINT_ADDR;
+	else if (cpuid_partno == CORTEX_M7)
+		if (target->part_id == 0x88c6U)
+			rom_location = IMXRT_11XX_ROM_FINGERPRINT_ADDR;
+		else
+			rom_location = IMXRT_10XX_ROM_FINGERPRINT_ADDR;
+	else
+		DEBUG_ERROR("Unknown core %04x\n", cpuid_partno);
+
+	uint32_t fingerprint = target_mem_read32(target, rom_location);
+	switch (fingerprint) {
+	case IMXRT_5XX_ROM_FINGERPRINT:
+		priv->chip_id = 500;
+		priv->flexspi_root = IMXRT_5XX_FLEXSPI1_BASE;
+		break;
+	case IMXRT_6XX_ROM_FINGERPRINT:
+		priv->chip_id = 600;
+		priv->flexspi_root = IMXRT_6XX_FLEXSPI1_BASE;
+		break;
+
+	case IMXRT_1011_ROM_FINGERPRINT:
+		priv->chip_id = 1011;
+		priv->flexspi_root = IMXRT_1011_FLEXSPI1_BASE;
+		break;
+	case IMXRT_102X_ROM_FINGERPRINT:
+		// The 1015 is actually a 1021.
+		priv->chip_id = 1021;
+		priv->flexspi_root = IMXRT_102X_FLEXSPI1_BASE;
+		break;
+	case IMXRT_105X_ROM_FINGERPRINT:
+		priv->chip_id = 1052;
+		priv->flexspi_root = IMXRT_105X_FLEXSPI1_BASE;
+		break;
+	case IMXRT_106X_ROM_FINGERPRINT:
+		// The 1042 is actually a 1062.
+		priv->chip_id = 1062;
+		priv->flexspi_root = IMXRT_106X_FLEXSPI1_BASE;
+		break;
+
+	case IMXRT_117X_ROM_FINGERPRINT:
+		priv->chip_id = 1176;
+		priv->flexspi_root = IMXRT_117X_FLEXSPI1_BASE;
+		break;
+	default:
+		DEBUG_TARGET("Unknown ROM fingerprint at %08x = %08x\n", rom_location, fingerprint);
+		break;
+	}
+
+	DEBUG_TARGET("Chip id: %d\n", priv->chip_id);
```

Comment:
You are storing chip_id as a uint16_t which is unsigned. `%d` is for *signed* integers. Please use `%u` here which is for unsigned integers. We'd replace or at least prefix the "Chip ID:" wording with the name of the function so we can be sure the message has come from the i.MXRT code.

```suggestion
	DEBUG_TARGET("%s: %u\n", __func__, priv->chip_id);
```